C error when passing by value instead of reference to `array.delete(voidptr)`.

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: linux, "Arch Linux"
Processor: 12 cpus, 64bit, little endian, Intel(R) Core(TM) i5-10600KF CPU @ 4.10GHz
CC version: cc (GCC) 11.1.0

getwd: /home/odd/source/v/vbird
vmodules: /home/odd/.vmodules
vroot: /home/odd/source/v/v
vexe: /home/odd/source/v/v/v
vexe mtime: 2021-11-06 18:13:39
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 9b00564.32b74dd

Git version: git version 2.33.0
Git vroot status: weekly.2021.44-25-g32b74dd3
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 333c14de

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg main.v`
```v
import gg
import gx
import rand

struct Game {
mut:
	gg         &gg.Context
	time       int
	birds      []Bird
	dead_birds []Bird
	pipes      []Pipe
}

// Initialize window and state, then run
fn main() {
	mut game := &Game{
		gg: 0
		pipes: pipes()
	}
	for _ in 0 .. 100 {
		game.birds << Bird{}
	}
	game.gg = gg.new_context(
		bg_color: gx.rgb(201, 206, 220)
		window_title: 'Flappy bird'
		frame_fn: frame
		user_data: game
		event_fn: on_event
	)
	game.gg.run()
}

// Bird logic
struct Bird {
mut:
	x        int = 256
	y        int = 1080 / 3
	size     int = 96
	velocity int
}

fn (mut bird Bird) update() {
	bird.velocity -= 1
	bird.y -= bird.velocity
}

// Pipe logic
struct Pipe {
mut:
	x int = 1920
	y int
	w int = 128
	h int
}

fn (mut pipe Pipe) update() {
	pipe.x -= 5
}

fn pipes() []Pipe {
	top := Pipe{
		y: 0
		h: rand.intn(128 + 1080 / 2)
	}
	bottom := Pipe{
		y: top.h + 128 * 3
		h: 1080
	}
	return [top, bottom]
}

// Collision
fn collides(bird Bird, pipe Pipe) bool {
	return bird.x < pipe.x + pipe.w && bird.x + bird.size > pipe.x && bird.y < pipe.y + pipe.h
		&& bird.size + bird.y > pipe.y
}

// Draw / update logic
fn frame(mut game Game) {
	game.gg.begin()

	// Timer update logic
	game.time++
	if game.time > 90 {
		game.pipes << pipes()
		game.time = 0
	}

	// Bird draw / update logic
	for mut bird in game.birds {
		bird.update()
		game.gg.draw_rect(bird.x, bird.y, bird.size, bird.size, gx.rgb(26, 24, 34))
		if game.pipes.any(collides(bird, it)) || bird.y < 0 || bird.y + bird.size > 1080 {
			index := game.birds.index(bird)	
			game.dead_birds << bird
			game.birds.delete(index)
		}
	}

	// Pipe draw / update logic
	for mut pipe in game.pipes {
		pipe.update()
		game.gg.draw_rect(pipe.x, pipe.y, pipe.w, pipe.h, gx.rgb(138, 198, 236))
	}

	game.gg.end()
}

// Key logic
fn on_event(e &gg.Event, mut game Game) {
	if e.typ == .key_down {
		match e.key_code {
			.escape { exit(0) }
			else {}
		}
	}
}

```

**What did you expect to see?**

Regular error. Occurs when passing bird by value instead of by reference.

**What did you see instead?**
```
==================
/tmp/v/main.18399998633036784434.tmp.c:21644: warning: assignment discards qualifiers from pointer target type
/tmp/v/main.18399998633036784434.tmp.c:24189: error: cannot convert 'struct main__Bird *' to 'struct main__Bird'
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/v/gen/c/array.v
+++ b/vlib/v/gen/c/array.v
@@ -762,6 +762,9 @@ fn (mut g Gen) gen_array_index(node ast.CallExpr) {
 	}
 	g.expr(node.left)
 	g.write(', ')
+	if node.args[0].expr.is_auto_deref_var() {
+		g.write('*')
+	}
 	g.expr(node.args[0].expr)
 	g.write(')')
 }

--- a/vlib/v/tests/for_in_mut_array_index_test.v
+++ b/vlib/v/tests/for_in_mut_array_index_test.v
@@ -1,0 +1,12 @@
+fn test_for_in_mut_array_index() {
+	mut arr := [1, 2, 3, 4, 5]
+	mut rets := []int{}
+
+	for mut val in arr {
+		inx := arr.index(val)
+		println(inx)
+		rets << inx
+	}
+
+	assert rets == [0, 1, 2, 3, 4]
+}

